core/lib/src/util/supabase_object.dart

```diff
@@ -94,8 +93,12 @@ class SupabaseQuery {
         notExtracted.add(JsonWithError(json, error));
       }
     }
-
-    return ExtractedSupabaseListResult(extracted, notExtracted);
+    if (notExtracted.isNotEmpty) {
+      // If some records could not be extracted, we throw an exception
+      // with the extracted records and the faulty records
+      throw ExtractedSupabaseListResult(extracted, notExtracted);
```

Comment:
Rename this to something like `ExtractingFailedException` and make sure that this type `extends Exception`